rust/stackable-cockpitd/src/handlers/ui.rs

```diff
@@ -16,16 +16,20 @@ pub fn router() -> Router {
 async fn ui() -> Html<&'static str> {
     Html(stackable_cockpit_web::INDEX_HTML)
 }
-async fn asset(Path(name): Path<String>) -> impl IntoResponse {
-    (
-        AppendHeaders([(
-            CONTENT_TYPE,
-            match name.split_once('.') {
-                Some((_, "js")) => HeaderValue::from_static("text/javascript"),
-                Some((_, "css")) => HeaderValue::from_static("text/css"),
-                _ => HeaderValue::from_static("application/octet-stream"),
-            },
-        )]),
-        stackable_cockpit_web::ASSETS[&name],
-    )
+
+/// Adds (or replaces) the Content-Type header with the type of the served asset.
+/// So far only javascript and css are supported, for all the other types
+/// `application/octet-stream` will be used.
```

Comment:
This is a detail of the function, but not its primary purpose. This should be either a subsection of the doccomment, or a regular comment inside of the function.